lib/ansible/modules/identity/kinit.py

```diff
@@ -0,0 +1,122 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'supported_by': 'community',
+    'status': ['preview']
+}
+
+DOCUMENTATION = '''
+---
+module: kinit
+version_added: "2.4"
+author: "Ali (@bincyber)"
+short_description: Obtain a Kerberos ticket-granting ticket.
+description:
+  - "This module is a wrapper around kinit to obtain a short-lived Kerberos ticket-granting ticket (TGT).
+  The M(expect) or M(shell) module could be used instead, however to ensure the principal's password is not logged,
+  the I(no_log: True) task attribute must be set resulting in hidden error messages and the loss of an audit trail.
+  By default, this module prevents the password for the authenticating principal from being shown or logged while
+  still displaying error messages and enabling the task to be logged for auditing purposes."
+options:
+  principal:
+    description:
+      - The principal to obtain a Kerberos ticket for. The principal can include the primary, instance, and realm. eg. primary/instance@REALM
+    required: yes
+    default: null
+  password:
+    description:
+      - The password used to authenticate to the Kerberos server.
+    required: yes
+    default: null
+  lifetime:
+    description:
+      - The lifetime of the Kerberos ticket.
+    required: false
+    default: 60
+requirements:
+  - kinit
+notes:
+  - "This module requires I(kinit) which needs to be installed on all target systems."
+'''
+
+EXAMPLES = '''
+# obtain a Kerberos ticket with a lifetime of 30 seconds
+- kinit:
+    principal: johndoe
+    password: supersecretpassword
+    lifetime: 30
+
+# obtain a Kerberos ticket for a principal that includes the realm
+- kinit:
+    principal: johndoe@REALM.EXAMPLE.COM
+    password: supersecretpassword
+'''
+
+RETURN = '''
+msg:
+    description: the status message describing what occurred
+    returned: always
+    type: string
+    sample: "Successfully obtained Kerberos ticket"
+
+rc:
+    description: the return code after executing kinit
+    returned: always
+    type: int
+    sample: 0
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from distutils.spawn import find_executable
+import shlex
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            principal=dict(required=True, type='str'),
+            password=dict(required=True, type='str', no_log=True),
+            lifetime=dict(required=False, type='int', default=60)
+        ),
+        required_together=[
+            ['principal', 'password']
+        ],
+        supports_check_mode=False,
+    )
+
+    principal = module.params['principal']
+    password = module.params['password']
+    lifetime = module.params.get('lifetime', 60)
+
+    kinit = find_executable('kinit')
```

Comment:
the AnsibleModule class has a method to find executables. Use that method:
kinit_cmd = module.get_bin_path("kinit", required=True)

Reply:
Done. Thanks